source/common/buffer/buffer_impl.h

```diff
@@ -358,49 +353,52 @@ class Slice {
     return num_pages * PageSize;
   }
 
-  static StoragePtr newStorage(uint64_t capacity, absl::optional<FreeListReference> free_list_opt) {
-    ASSERT(sliceSize(default_slice_size_) == default_slice_size_,
-           "default_slice_size_ incompatible with sliceSize()");
-    ASSERT(sliceSize(capacity) == capacity,
-           "newStorage should only be called on values returned from sliceSize()");
-    ASSERT(!free_list_opt.has_value() || &free_list_opt->free_list_ == &free_list_);
-
-    StoragePtr storage;
-    if (capacity == default_slice_size_ && free_list_opt.has_value()) {
-      FreeListType& free_list = free_list_opt->free_list_;
-      if (!free_list.empty()) {
-        storage = std::move(free_list.back());
-        ASSERT(storage != nullptr);
-        ASSERT(free_list.back() == nullptr);
-        free_list.pop_back();
-        return storage;
-      }
+  static constexpr uint32_t free_list_max_ = 2 * Buffer::Reservation::MAX_SLICES_;
```

Comment:
Why 2x? That increases the size of several structs by quite a bit.

Reply:
I've made various attempts before. This is a small omission when I cleaned up the code.